lib/external-parsers/base.ts

```diff
@@ -0,0 +1,98 @@
+import fs from 'fs';
+import path from 'path';
+
+import { TypicalExecutionFunc, UnprocessedExecResult } from '../../types/execution/execution.interfaces';
+import { IParseFilters } from '../../types/features/filters.interfaces';
+import { maskRootdir } from '../utils';
+
+import { IExternalParser, IParsedAsmResult } from './external-parser.interface';
+
+const starterScriptName = 'dump-and-parse.sh';
+
+export class ExternalParserBase implements IExternalParser {
+    private objdumperPath: string;
+    private parserPath: string;
+    private execFunc: TypicalExecutionFunc;
+    private compilerInfo;
+    private envInfo;
+
+    constructor(compilerInfo, envInfo, execFunc: TypicalExecutionFunc) {
+        this.compilerInfo = compilerInfo;
+        this.envInfo = envInfo;
+        this.objdumperPath = compilerInfo.objdumper;
+        this.parserPath = compilerInfo.externalparser.props('exe', '');
+        this.execFunc = execFunc;
+    }
+
+    private getParserArguments(filters: IParseFilters, fromStdin: boolean): string[] {
+        const parameters = ['-plt'];
+
+        if (fromStdin) parameters.push('-stdin');
+        if (filters.binary) parameters.push('-binary');
+        if (filters.labels) parameters.push('-unused_labels');
+        if (filters.directives) parameters.push('-directives');
+        if (filters.commentOnly) parameters.push('-comment_only');
+        if (filters.trim) parameters.push('-whitespace');
+        if (filters.libraryCode) parameters.push('-library_functions');
+        if (filters.dontMaskFilenames) parameters.push('-dont_mask_filenames');
+
+        return parameters;
+    }
+
+    private getObjdumpStarterScriptContent(filters: IParseFilters) {
+        const parserArgs = this.getParserArguments(filters, true);
+
+        return '#!/bin/bash\n' +
+            `OBJDUMP=${this.objdumperPath}\n` +
+            `ASMPARSER=${this.parserPath}\n` +
+            `$OBJDUMP "$@" | $ASMPARSER ${parserArgs.join(' ')}\n`;
```

Comment:
this works but it's a bit subtle that `parserArgs` must not contain anything that isn't shell-safe. I wonder if we can avoid the shell involvement...but not for this PR :)

Reply:
`parserArgs` should always be something that we ourselves compose and not the user, so I don't think this should be a problem